apps/rush-lib/src/cli/actions/AddAction.ts

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as os from 'os';
+
+import {
+  CommandLineFlagParameter,
+  CommandLineStringParameter
+} from '@microsoft/ts-command-line';
+
+import { RushConfigurationProject } from '../../api/RushConfigurationProject';
+import { BaseRushAction } from './BaseRushAction';
+import { RushCommandLineParser } from '../RushCommandLineParser';
+import { DependencyIntegrator, SemVerStyle } from '../../logic/DependencyIntegrator';
+
+export class AddAction extends BaseRushAction {
+  private _exactFlag: CommandLineFlagParameter;
+  private _caretFlag: CommandLineFlagParameter;
+  private _devDependencyFlag: CommandLineFlagParameter;
+  private _makeConsistentFlag: CommandLineFlagParameter;
+  private _noInstallFlag: CommandLineFlagParameter;
+  private _packageName: CommandLineStringParameter;
+  private _versionSpecifier: CommandLineStringParameter;
+
+  constructor(parser: RushCommandLineParser) {
+    const documentation: string[] = [
+      'Blah.'
+    ];
+    super({
+      actionName: 'add',
+      summary: 'Adds a dependency to the package.json and runs rush upgrade.',
+      documentation: documentation.join(os.EOL),
+      safeForSimultaneousRushProcesses: false,
+      parser
+    });
+  }
+
+  public onDefineParameters(): void {
+    this._packageName = this.defineStringParameter({
+      parameterLongName: '--package',
+      parameterShortName: '-p',
+      required: true,
+      argumentName: 'PACKAGE_NAME',
+      description: '(Required) The name of the package which should be added as a dependency'
+    });
```

Comment:
This is clunky. How hard would it be to add an "anonymous" argument to TS-command-line?

Reply:
Pete and I discussed this, but it will be a pretty big change in ts-command-line. We decided to just go with the normal way for now, then potentially change it later.

---
In reply to: [220407271](https://github.com/Microsoft/web-build-tools/pull/843#discussion_r220407271) [](ancestors = 220407271)